### ISSUE TITLE
question!

### ISSUE THREAD

**Issue by aif2017** [zzh8829/yolov3-tf2#8]
hi why you have done this?

def YoloLoss(anchors, classes=80, ignore_thresh=0.5):
      def yolo_loss(y_true, y_pred):
              .

return yolo_loss

**Comment by zzh8829** [comment]
This is because keras calls loss functions with loss(y_true, y_pred). But we need more parameters than just y_true, y_pred (for example, anchors, classes etc) Thats why we wrap around the real yolo_loss function with the YoloLoss function to create a [function closure](https://en.wikipedia.org/wiki/Closure_(computer_programming)) with extra parameters.

**Comment by aif2017** [comment]
thx

**Comment by zzh8829** [comment]
Ya training is only limited to 416 for now, but it's should be easy to extend to other sizes.
However, after the model is trained, it will work on all resolutions
